### PR TITLE
Feature/auth

### PR DESCRIPTION
**Pull request by billxinli**
## What Changed & Why
Added new Initializer option of passing in username and password (direct pass through to axios) for HTTP Basic Auth

ie:
`var jsonApi = new JsonApi({apiUrl: 'https://example.com', auth: {username: 'secret', password: 'word'} })`

And the `HTTP_BASIC_AUTH` middleware will pass this to axios and request should have http auth headers attached it it

## Testing
`npm test`

## People
@Emerson 
@themarkappleby

### CODE REVIEW

**Comment by Emerson** [comment]
Awsome

File: test/api/api-test.js [hunk 1]
```diff
@@ -78,6 +78,26 @@ describe('JsonApi', () => {
       })
     })
 
+    it('should allow users to add HTTP Basic Auth options', (done) => {
+      jsonApi = new JsonApi({apiUrl: 'http://myapi.com', auth: {username: 'admin', password: 'cheesecake'}})
+
+      jsonApi.define('foo', {title: ''})
+
+      let inspectorMiddleware = {
```
Comment (Emerson) [inline test/api/api-test.js]: I love this. I love how our tests are using our own middleware layer... so meta.

**Comment by billxinli** [comment]
@Emerson Totally separate issue, but wondering if a changelog should be created, similar to: https://github.com/sequelize/sequelize/blob/master/changelog.md

**Comment by Emerson** [comment]
Probably... I'd be fine adding that – I guess we just maintain that kinda stuff by hand?

**Comment by billxinli** [comment]
Yep. Add a feature, prepend to the changelog.

It made sense with the sequelize library, since it is a very complex
library with a lot of moving parts. I wonder if it also makes sense here.

**Pull request merged**